resources/resource_factories/create/create.go

```diff
@@ -126,3 +145,161 @@ func (c *Client) FromString(targetPath, content string) (resource.Resource, erro
 			})
 	})
 }
+
+// FromRemote expects one or n-parts of a URL to a resource
+// If you provide multiple parts they will be joined together to the final URL.
+func (c *Client) FromRemote(uri string, options map[string]interface{}) (resource.Resource, error) {
+	rURL, err := url.Parse(uri)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse URL for resource %s", uri)
+	}
+
+	resourceID := helpers.HashString(uri, options)
+
+	return c.rs.ResourceCache.GetOrCreate(resources.ResourceCacheKey(resourceID), func() (resource.Resource, error) {
```

Comment:
I take that back, just use the `resourceID` as is for the key for now (I will fix this cache situation in a bit).

Reply:
You mean going back to:
```
resourceID := helpers.MD5String(uri)
```

I do think it's good/a must to have optional headers, request method, etc. included in the cache key.
Any reason you think it's better to go back to just the URL?